instance/tests/management/test_reprovision_buckets.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (C) 2015-2018 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Instance - reprovision_buckets unit tests
+"""
+from unittest.mock import patch
+
+from django.core.exceptions import FieldDoesNotExist
+from django.core.management import call_command
+from django.test import TestCase
+from django.utils.six import StringIO
+from testfixtures import LogCapture
+
+from instance.models.openedx_instance import OpenEdXInstance
+
+
+class ReprovisionBucketsTestCase(TestCase):
+    """
+    Test cases for the `reprovision_buckets` management command.
+    """
+
+    def test_no_instances(self):
+        """
+        Verify that the command correctly notifies the user that there are no instances for migration.
+        """
+        with LogCapture() as captured_logs:
+            call_command(
+                'reprovision_buckets',
+                stdout=StringIO(),
+            )
+            # Verify the logs
+            self.assertIn(
+                'Found "0" active instances',
+                [l[2] for l in captured_logs.actual()])
+
+    @patch('instance.models.mixins.storage.S3BucketInstanceMixin._enable_bucket_versioning')
+    @patch('instance.models.mixins.storage.S3BucketInstanceMixin._update_bucket_lifecycle')
+    @patch('instance.models.mixins.storage.S3BucketInstanceMixin._update_bucket_cors')
+    @patch('instance.models.mixins.storage.S3BucketInstanceMixin._perform_create_bucket')
+    @patch('instance.models.mixins.storage.S3BucketInstanceMixin._update_iam_policy')
+    @patch('instance.models.mixins.storage.S3BucketInstanceMixin.create_iam_user')
+    def test_migrate(self,
+                     mock_create_iam_user,
+                     mock_update_iam,
+                     mock_create_bucket,
+                     mock_update_cors,
+                     mock_update_lifecycle,
+                     mock_enable_versioning):
+        """
+        Verify that the command correctly reprovision the bucket for an instance.
+        """
+        OpenEdXInstance.objects.create(
+            sub_domain='test', name='test instance', storage_type=OpenEdXInstance.S3_STORAGE)
+        with LogCapture() as captured_logs:
+            call_command(
+                'reprovision_buckets',
+                stdout=StringIO(),
+            )
+            # Verify the logs
```

Comment:
I think these tests should be outside the context manager, to ensure `captured_logs` isn't being modified after the call to `call_command()`.

Reply:
The command must be inside the context manager as it's actually verifying the output o the command.

Reply:
Sorry, I put the comment in a weird place. The `call_command` command needs to be inside the context manager to capture its output. But I think the assertion can be outside the context manager, because `captured_logs` is still in scope, but then `LogCapture` isn't still appending new logs to it?

Reply:
👍